Issue 1259/1344: Views

simplypixi opened the pull request:
This fixes #1259 and #1344.

simplypixi commented:
@sanmont3drepo , @carmenfan I removed re-loading previous model, but I noticed two bugs, which appear on staging too:

1. When I load Lego Architecture model, go to Teamspaces and then load Lego Federation model the viewer throws some errors related to the pins. It can't to remove the pins, because they exists in pins map but not in unity context.
![image](https://user-images.githubusercontent.com/5683490/51836900-18884000-2302-11e9-91d4-d8c40b60ef0e.png)

2. When I go to Teamspace while the model is loading, the viewer doesn't cancel request/model rendering, so it's appears in the Teamspaces background. I think Pavol reported exactly the same bug, but it seems to be an issue on viewer side (e.g $destroy in viewer.component never calls).

What do you think about it?

carmenfan commented:
> @sanmont3drepo , @carmenfan I removed re-loading previous model, but I noticed two bugs, which appear on staging too:
> 
> 1. When I load Lego Architecture model, go to Teamspaces and then load Lego Federation model the viewer throws some errors related to the pins. It can't to remove the pins, because they exists in pins map but not in unity context.

I'm guessing this only happens on initialisation? It's probably worth taking a look during issue migration, as the pins should be consistent with unity provided the model has started loading. I think I've seen this in the pass and it doesn't really affect the functionality as the pins were loaded after the model started loading.

> 2. When I go to Teamspace while the model is loading, the viewer doesn't cancel request/model rendering, so it's appears in the Teamspaces background. I think Pavol reported exactly the same bug, but it seems to be an issue on viewer side (e.g $destroy in viewer.component never calls).
> 
> What do you think about it?

This is a known issue. It's not to do with the component not getting destroyed. It's API requests that has been fired in unity, and before they come back reset has been called. So the whole dataset has been wiped in Unity, but then we get the callback.

We know this happens for some time now and if we were too address it we need to change i in Unity, so I wouldn't worry too much about it. (as long as it's clearing the viewer if you exit after the model has been loaded

simplypixi commented:
@carmenfan @sanmont3drepo @dleszcz Rebase and ready for review ;)

sanmont3drepo commented on frontend/components/index.ts:
```diff
@@ -60,6 +60,4 @@ import './utils/js/utils.filter';
 import './viewer/js/multi-select.service';
 import './viewer/js/viewer.component';
 import './viewer/js/viewer.service';
-import './viewpoints/js/viewpoints.component';
```
Should delete the viewpoints angular component folder also.

simplypixi replied:
done

sanmont3drepo commented on frontend/modules/viewer/viewer.sagas.ts:
```diff
@@ -30,7 +31,7 @@ export function* waitForViewer() {
 		yield ViewerService.initialised.promise;
 	} catch (error) {
-		console.error(error);
+		yield put(DialogActions.showErrorDialog('initialise', 'viewer', error));
```
- If the error is an enpoint error should use the DialogActions.showEndpointErrorDialog action instead. If it is a string its ok. Please check every call.

sanmont3drepo commented on frontend/routes/viewer/components/views/components/viewItem/viewItem.component.tsx:
```diff
@@ -0,0 +1,119 @@
+	onSaveEdit: (values) => void;
+	onDelete?: (event) => void;
+	onOpenEditMode?: () => void;
+	handleClick?: (viewpoint) => void;
```
- Can we rename **handleClick** to **onClick** ?

simplypixi replied:
Yes, definitely. I don't know why it was named like this. :/

simplypixi commented:
> I think there is a bug with the resizing of the card.
> When you have an empty list of views the height is small and stays small until you refresh the page.
> 
> Now:
> ![resizing bad](https://user-images.githubusercontent.com/42032433/51992003-25dd3000-24a4-11e9-9485-f21419d22854.PNG)
> Empty list:
> ![resizing bad2](https://user-images.githubusercontent.com/42032433/51992015-2f669800-24a4-11e9-885b-4c3dad687c1c.PNG)
> 
> How it should look:
> ![resizing ok](https://user-images.githubusercontent.com/42032433/51992045-39889680-24a4-11e9-9eee-0a438c03d218.PNG)

Fixed ;)

sanmont3drepo commented:
I think everything is ok, can I merge this @carmenfan ?

carmenfan commented:
Let me put this through the @paloknapo test first :)

carmenfan commented:
@simplypixi 
@paloknapo says everything works, but the icons should be in one row:
![image](https://user-images.githubusercontent.com/11945337/52137292-24496e80-2642-11e9-94b9-e55c19cb5e72.png)
![image](https://user-images.githubusercontent.com/11945337/52137305-29a6b900-2642-11e9-93d4-424b23a1eb0e.png)

Also the icons should be in grey, like this:
![image](https://user-images.githubusercontent.com/11945337/52137323-32978a80-2642-11e9-9a21-0dbff30f2423.png)

simplypixi commented:
> @simplypixi
> @paloknapo says everything works, but the icons should be in one row:
> ![image](https://user-images.githubusercontent.com/11945337/52137292-24496e80-2642-11e9-94b9-e55c19cb5e72.png)
> ![image](https://user-images.githubusercontent.com/11945337/52137305-29a6b900-2642-11e9-93d4-424b23a1eb0e.png)
> 
> Also the icons should be in grey, like this:
> ![image](https://user-images.githubusercontent.com/11945337/52137323-32978a80-2642-11e9-9a21-0dbff30f2423.png)

I changed these icons on Risks branch about a week ago. I think this PR was created before the views got new icons' look. Move this fix here? ;)

simplypixi commented:
> > @simplypixi
> > @paloknapo says everything works, but the icons should be in one row:
> > ![image](https://user-images.githubusercontent.com/11945337/52137292-24496e80-2642-11e9-94b9-e55c19cb5e72.png)
> > ![image](https://user-images.githubusercontent.com/11945337/52137305-29a6b900-2642-11e9-93d4-424b23a1eb0e.png)
> > Also the icons should be in grey, like this:
> > ![image](https://user-images.githubusercontent.com/11945337/52137323-32978a80-2642-11e9-9a21-0dbff30f2423.png)
> 
> I changed these icons on Risks branch about a week ago. I think this PR was created before the views got new icons' look. Move this fix here? ;)

Fixed ;)

simplypixi commented:
> @simplypixi
> 
> On the Angular side, when there's a deleted group (or closed issues / resolved risks) happened on another client and you get the update via the chat service, We get this little message before the item disappears. Can we do the same for views?
> 
> ![1](https://user-images.githubusercontent.com/11945337/52208301-082f1280-2878-11e9-98b0-52232a387119.gif)

Sure. ;) Should it look the same?

simplypixi commented:
@carmenfan Fixed and rebased ;)

carmenfan commented:
@simplypixi looking good :)

But this seems to happen still?

> Selected view should unselect when you add a new view:
> ![1](https://user-images.githubusercontent.com/11945337/52207860-a91cce00-2876-11e9-80fb-8987222bbfa9.gif)

simplypixi commented:
> @simplypixi looking good :)
> 
> But this seems to happen still?
> 
> > Selected view should unselect when you add a new view:
> > ![1](https://user-images.githubusercontent.com/11945337/52207860-a91cce00-2876-11e9-80fb-8987222bbfa9.gif)

Fixed ;)
